src/server/ApiDispatchers.cpp

```diff
@@ -34,6 +34,15 @@ static DWORD TraceGetThreadId(CONSOLE_API_MSG* const m)
     return p ? p->dwThreadId : 0;
 }
 
+template<typename T>
+constexpr T saturate(auto val)
+{
+    constexpr auto min = std::numeric_limits<T>::min();
+    constexpr auto max = std::numeric_limits<T>::max();
+#pragma warning(suppress : 4267) // '...': conversion from '...' to 'T', possible loss of data
+    return val < min ? min : (val > max ? max : val);

```

Comment:
If `typeof(val)` is `T`, these checks are UB, right? Since the compiler can probably optimize out `T(x) < T(min)` as it will always be false?

should we assert that the range of `T` is wider than the range of `typeof(val)`?

Reply:
meh, it's probably fine. `saturate<T>(T)` _should_ just `return T;` afterall. These checks being eliminated makes that true!

Reply:
I don't think it would be UB, it would just be always false (as you said).
I haven't really thought this function through yet, tbh. It's basically a "prototype" and that's also why I left it there and not in til.